backend/app/routers/metadata_datasets.py

```diff
@@ -229,8 +229,9 @@ async def update_dataset_metadata(
 
         md = await MetadataDB.find_one(*query)
         if md is not None:
-            await index_dataset(es, dataset, update=True)
-            return await patch_metadata(md, content, es)
+            patched_metadata = await patch_metadata(md, content, es)
+            await index_dataset(es, DatasetOut(**dataset.dict()), update=True)
+            return patched_metadata
```

Comment:
There was a bug here. The metadata is updated after indexing hence the updated metadata never reflected